kolibri/core/assets/test/views/NotificationsRoot.spec.js

```diff
@@ -0,0 +1,87 @@
+import { shallowMount } from '@vue/test-utils';
+import NotificationsRoot from '../../src/views/NotificationsRoot.vue';
+import { coreStoreFactory as makeStore } from '../../src/state/store';
+
+function makeWrapper(options = {}) {
+  const store = makeStore();
+  const wrapper = shallowMount(NotificationsRoot, {
+    store,
+    ...options,
+    computed: {
+      notAuthorized: () => false,
+      showNotification: () => false,
+      mostRecentNotification: () => {
+        return {
+          id: 1,
+          title: 'title',
+          msg: 'notification',
+          linkText: 'linktext',
+          linkUrl: 'url',
+        };
+      },
+      ...(options.computed || {}),
+    },
+  });
+  return { wrapper, store, ...options };
+}
+
+describe('NotificationsRoot', function() {
+  it('smoke test', () => {
+    const { wrapper } = makeWrapper();
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  describe('when loaded', function() {
+    it('if user is authorized and there is no error, main div for displaying <slot> should be displayed', async () => {
+      const { wrapper, store } = makeWrapper();
+      store.state.core.loading = false;
+      await wrapper.vm.$nextTick();
+
+      expect(wrapper.find('[data-test="main"]').exists()).toBeTruthy();
+      expect(wrapper.findComponent({ name: 'AuthMessage' }).exists()).toBeFalsy();
+      expect(wrapper.findComponent({ name: 'AppError' }).exists()).toBeFalsy();
+    });
+
+    it('if user is not authorized, AuthMessage component should be rendered', async () => {
+      const { wrapper, store } = makeWrapper({ computed: { notAuthorized: () => true } });
```

Comment:
This test, as implemented right now, doesn't really test this scenario. Assume that over time we'd make a mistake in code of [the `notAuthorized` computed property ](https://github.com/learningequality/kolibri/pull/9128/files#diff-4a205996a86d148f2aeb0e7a84a2f6ea6325fe25f0f548eae3230a4390abdce4R83), for example:

```javascript
notAuthorized() {
  if (
    this.error &&
    this.error.response &&
    this.error.response.status &&
    this.error.response.status == 403
  ) {
    return false; // should be `return true`
  }
  return !this.authorized;
},
```

Even though `AuthMessage` wouldn't get displayed after we got 403 response, this test would still pass because `notAuthorized` is set to `true` no matter of what due to how the test wrapper is created `makeWrapper({ computed: { notAuthorized: () => true } })`, giving us the false impression that things work as expected.

I see this pattern in couple of other tests of this suite too. I'd suggest removing anything related to `computed` from the whole test suite and mocking rather things that are coming from outside of the `NotificationsRoot` (store state etc.) instead of mocking computed properties, similarly to how you did it [on lines 57-58](https://github.com/learningequality/kolibri/pull/9128/files#diff-0ad00cb2ee84c775bb92409d2ea55408ffa548d7bb3546116225a01dce333913R58).

I wrote more generally about problems of reaching into internal implementation in tests in https://github.com/learningequality/kolibri/pull/9065#pullrequestreview-870732578 if some more information would help.